okhttp/src/main/java/com/squareup/okhttp/Call.java

```diff
@@ -270,7 +272,20 @@ Response getResponse(Request request, boolean forWebSocket) throws IOException {
       try {
         engine.sendRequest();
         engine.readResponse();
+      } catch (RequestException e) {
+        // The attempt to interpret the request failed. Give up.
```

Comment:
Can we make request validation more eager? What happens if a redirect target is malformed?